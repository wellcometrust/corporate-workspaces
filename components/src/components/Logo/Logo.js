// @flow
import React from 'react';
import styled from 'styled-components';

type Props = {|
  title?: string,
  fill?: string,
  className?: string,
  data: {
    viewBox: string,
    fillRule: string,
    path: string
  }
|};

const defaultProps = {
  title: '',
  fill: '#fff',
  className: ''
};

const StyledLogo = styled.svg`
  height: 100%;
`;

const Logo = ({
  title,
  fill,
  className,
  data: { viewBox, fillRule, path },
  ...otherProps
}: Props) => (
  <StyledLogo
    viewBox={viewBox}
    className={className}
    role="img"
    aria-labelledby="logo-title"
    preserveAspectRatio="xMinYMid"
    {...otherProps}
  >
    {title ? <title id="logo-title">{title}</title> : ''}
    <path d={path} fill={fill} fillRule={fillRule} />
  </StyledLogo>
);

Logo.defaultProps = defaultProps;

export default Logo;
