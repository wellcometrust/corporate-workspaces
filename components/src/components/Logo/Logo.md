# Logo

Wellcome official logo variations (small, medium and large).
