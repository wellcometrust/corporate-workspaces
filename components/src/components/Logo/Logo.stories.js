import React from 'react';
import { storiesOf } from '@storybook/react';
import Logo from './Logo';
import LogoReadme from './Logo.md';
import logoData from './constants';

const stories = storiesOf('Components|Logo', module);

stories
  .addParameters({
    readme: {
      sidebar: LogoReadme
    }
  })
  .add('small', () => {
    const logoSmall = logoData.small;
    return <Logo data={logoSmall} fill="black" />;
  })
  .add('medium', () => {
    const logoMedium = logoData.medium;
    return <Logo data={logoMedium} fill="black" />;
  })
  .add('large', () => {
    const logoLarge = logoData.large;
    return <Logo data={logoLarge} fill="black" />;
  });
