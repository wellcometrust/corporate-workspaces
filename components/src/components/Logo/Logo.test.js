import React from 'react';
import { shallow } from 'enzyme';

import Logo from './Logo';
import logoData from './constants';

describe('<Logo />', () => {
  it('renders a small logo', () => {
    const output = shallow(<Logo data={logoData.small} />);

    expect(output).toMatchSnapshot();
  });

  it('renders a medium logo', () => {
    const output = shallow(<Logo data={logoData.medium} />);

    expect(output).toMatchSnapshot();
  });

  it('renders a large logo', () => {
    const output = shallow(<Logo data={logoData.large} />);

    expect(output).toMatchSnapshot();
  });

  it('renders as a black logo', () => {
    const output = shallow(<Logo data={logoData.large} fill="black" />);

    expect(output).toMatchSnapshot();
  });
});
