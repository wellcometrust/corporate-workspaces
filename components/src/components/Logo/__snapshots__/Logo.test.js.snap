// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders a large logo 1`] = `ShallowWrapper {}`;

exports[`<Logo /> renders a medium logo 1`] = `ShallowWrapper {}`;

exports[`<Logo /> renders a small logo 1`] = `ShallowWrapper {}`;

exports[`<Logo /> renders as a black logo 1`] = `ShallowWrapper {}`;
